components/card/index.jsx

```diff
@@ -0,0 +1,47 @@
+import { createElement } from "complate-stream";
+
+export function Card(_, ...children) {
+	return <div class="card">{ children }</div>;
+}
+
+export function CardImg({ src, alt }, ...children) {
+	return <img class="card-img-top" src={ src } alt={ alt }/>;
```

Comment:
There are three classes we could use here: `card-img`, `card-img-top`, and `card-img-bottom`. The only differences are their `border-radius` properties. Maybe we could choose one of them with an optional `placement` attribute, and use `card-img` if it’s missing?

The downside is that I believe the `card-img` is intended to fill the entire card, with _another_ element overlaid on top of it (indicated by `card-img-overlay`).
Maybe if we have any `children` we could wrap them in a `div` element with `card-img-overlay`?

Reply:
My thinking was to have the `CardImg` being able to fill the role of `card-img`, `card-img-top` and `card-img-bottom` by a placement attribute.

About that overlay – we could do the following:

```jsx
<CardImg src="fancy.jpg">
  <CardTitle>Fancy Picture</CardTitle>
</CardImg>
```

outputs:

```html
<img class="card-img" src="fancy.jpg">
<div class="card-img-overlay">
  <h5 class="card-title">Fancy Picture</h5>
</div>
```

Or is this too much magic? Then we could simply do the following:

```jsx
<CardImg src="fancy.jpg" />
<CardImgOverlay>
  <CardTitle>Fancy Picture</CardTitle>
</CardImgOverlay>
```

Reply:
Yes, that (the former variant) is exactly what I meant by
> Maybe if we have any `children` we could wrap them in a `div` element with `card-img-overlay`?

I’m in favour.